daemon/docker_watcher.go

```diff
@@ -280,9 +270,11 @@ func (d *Daemon) handleCreateContainer(id string, retry bool) {
 		}
 
 		var orchLabelsModified bool
-		containerCopy, ok := d.retrieveWorkingContainerCopy(id)
+		d.containersMU.RLock()
+		containerCopy, ok := d.containers[id]
+		d.containersMU.RUnlock()
 		if ok {
-			if orchLabelsModified = updateOrchLabels(&containerCopy, lbls); !orchLabelsModified {
+			if orchLabelsModified = updateOrchLabels(containerCopy, lbls); !orchLabelsModified {
```

Comment:
I made a whole bunch of changes on this PR and afterwards I split them into logic commits. The `containerCopy` is not guarantee that only a go subroutine modifies the container object at this time. I can move the `d.containerMU.RUnlock()` after the `updateOrchLabels` as a `Unlock()` on this commit. That way there won't be "bad" commits, sounds good?